### PR TITLE
fix: Fix for #59 and improved typing

### PR DESCRIPTION
**Pull request by DickSmith**
Fix for #59 and improved typing
```
Fatal Exception: com.tns.NativeScriptException
Calling js method onCreateView failed Error: com.facebook.imagepipeline.request.ImageRequestBuilder$BuilderException: Invalid request builder: Resource URI must not be empty com.facebook.imagepipeline.request.ImageRequestBuilder.validate(ImageRequestBuilder.java:277) com.facebook.imagepipeline.request.ImageRequestBuilder.build(ImageRequestBuilder.java:253) com.tns.Runtime.callJSMethodNative(Native Method) com.tns.Runtime.dispatchCallJSMethodNative(Runtime.java:1100) com.tns.Runtime.callJSMethodImpl(Runtime.java:982) com.tns.Runtime.callJSMethod(Runtime.java:969) com.tns.Runtime.callJSMethod(Runtime.java:953) com.tns.Runtime.callJSMethod(Runtime.java:945) com.tns.FragmentClass.onCreateView(FragmentClass.java:45) android.app.Fragment.performCreateView(Fragment.java:2220) android.app.FragmentManagerImpl.moveToState(FragmentManager.java:973) android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1148) android.app.BackStackRecord.run(BackStackRecord.java:793) android.app.FragmentManagerImpl.execPendingActions(FragmentManager.java:1535) android.app.FragmentManagerImpl$1.run(FragmentManager.java:482) android.os.Handler.handleCallback(Handler.java:739) android.os.Handler.dispatchMessage(Handler.java:95) android.os.Looper.loop(Looper.java:148) android.app.ActivityThread.main(ActivityThread.java:5417) java.lang.reflect.Method.invoke(Native Method) com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:742) com.android.internal.os.ZygoteInit.main(ZygoteInit.java:632) File: "<embedded>, line: 36, column: 1105305 StackTrace: Frame: function:'t.initImage', file:'<embedded>', line: 36, column: 1105306 Frame: function:'t.initDrawee', file:'<embedded>', line: 36, column: 1104279 Frame: function:'t.initNativeView', file:'<embedded>', line: 36, column: 1102653 Frame: function:'t.setNativeView', file:'<embedded>', line: 36, column: 1286625 Frame: function:'t._setupUI', file:'<embedded>', line: 36, column: 1286119 Frame: function:'', file:'<embedded>', line: 36, column: 1286374 Frame: function:'t.eachChildView', file:'<embedded>', line: 36, column: 1464842 Frame: function:'n.eachChild', file:'<embedded>', line: 36, column: 1273517 Frame: function:'t._setupUI', file:'<embedded>', line: 36, column: 1286350 Frame: function:'', file:'<embedded>', line: 36, column: 1286374 Frame: function:'t.eachChildView', file:'<embedded>', line: 36, column: 1464842 Frame: function:'n.eachChild', file:'<embedded>', line: 36, column: 1273517 Frame: function:'t._setupUI', file:'<embedded>', line: 36, column: 1286350 Frame: function:'', file:'<embedded>', line: 36, column: 1286374 Frame: function:'t.eachChildView', file:'<embedded>', line: 36, column: 1464842 Frame: function:'n.eachChild', file:'<embedded>', line: 36, column: 1273517 Frame: function:'t._setupUI', file:'<embedded>', line: 36, column: 1286350 Frame: function:'', file:'<embedded>', line: 36, column: 1286374 Frame: function:'t.eachChildView', file:'<embedded>', line: 36, column: 628054 Frame: function:'n.eachChild', file:'<embedded>', line: 36, column: 1273517 Frame: function:'t._setupUI', file:'<embedded>', line: 36, column: 1286350 Frame: function:'', file:'<embedded>', line: 36, column: 1286374 Frame: function:'t.eachChildView', file:'<embedded>', line: 36, column: 1464842 Frame: function:'n.eachChild', file:'<embedded>', line: 36, column: 1273517 Frame: function:'t._setupUI', file:'<embedded>', line: 36, column: 1286350 Frame: function:'', file:'<embedded>', line: 36, column: 1286374 Frame: function:'t.eachChildView', file:'<embedded>', line: 36, column: 628054 Frame: function:'t.eachChildView', file:'<embedded>', line: 36, column: 1448752 Frame: function:'n.eachChild', file:'<embedded>', line: 36, column: 1273517 Frame: function:'t._setupUI', file:'<embedded>', line: 36, column: 1286350 Frame: function:'t._addViewCore', file:'<embedded>', line: 36, column: 1284245 Frame: function:'t._addView', file:'<embedded>', line: 36, column: 1283968 Frame: function:'e.onCreateView', file:'<embedded>', line: 36, column: 560440 Frame: function:'t.onCreateView', file:'<embedded>', line: 36, column: 1316946
com.tns.Runtime.callJSMethodNative (Runtime.java)
com.tns.Runtime.dispatchCallJSMethodNative (Runtime.java:1100)
com.tns.Runtime.callJSMethodImpl (Runtime.java:982)
com.tns.Runtime.callJSMethod (Runtime.java:969)
com.tns.Runtime.callJSMethod (Runtime.java:953)
com.tns.Runtime.callJSMethod (Runtime.java:945)
com.tns.FragmentClass.onCreateView (FragmentClass.java:45)
android.app.Fragment.performCreateView (Fragment.java:2220)
android.app.FragmentManagerImpl.moveToState (FragmentManager.java:973)
android.app.FragmentManagerImpl.moveToState (FragmentManager.java:1148)
android.app.BackStackRecord.run (BackStackRecord.java:793)
android.app.FragmentManagerImpl.execPendingActions (FragmentManager.java:1535)
android.app.FragmentManagerImpl$1.run (FragmentManager.java:482)
android.os.Handler.handleCallback (Handler.java:739)
android.os.Handler.dispatchMessage (Handler.java:95)
android.os.Looper.loop (Looper.java:148)
android.app.ActivityThread.main (ActivityThread.java:5417)
java.lang.reflect.Method.invoke (Method.java)
com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:742)
com.android.internal.os.ZygoteInit.main (ZygoteInit.java:632)
```

```
Caused by com.facebook.imagepipeline.request.ImageRequestBuilder$BuilderException
Invalid request builder: Resource URI must not be empty
com.facebook.imagepipeline.request.ImageRequestBuilder.validate (ImageRequestBuilder.java:277)
com.facebook.imagepipeline.request.ImageRequestBuilder.build (ImageRequestBuilder.java:253)
com.tns.Runtime.callJSMethodNative (Runtime.java)
com.tns.Runtime.dispatchCallJSMethodNative (Runtime.java:1100)
com.tns.Runtime.callJSMethodImpl (Runtime.java:982)
com.tns.Runtime.callJSMethod (Runtime.java:969)
com.tns.Runtime.callJSMethod (Runtime.java:953)
com.tns.Runtime.callJSMethod (Runtime.java:945)
com.tns.FragmentClass.onCreateView (FragmentClass.java:45)
android.app.Fragment.performCreateView (Fragment.java:2220)
android.app.FragmentManagerImpl.moveToState (FragmentManager.java:973)
android.app.FragmentManagerImpl.moveToState (FragmentManager.java:1148)
android.app.BackStackRecord.run (BackStackRecord.java:793)
android.app.FragmentManagerImpl.execPendingActions (FragmentManager.java:1535)
android.app.FragmentManagerImpl$1.run (FragmentManager.java:482)
android.os.Handler.handleCallback (Handler.java:739)
android.os.Handler.dispatchMessage (Handler.java:95)
android.os.Looper.loop (Looper.java:148)
android.app.ActivityThread.main (ActivityThread.java:5417)
java.lang.reflect.Method.invoke (Method.java)
com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:742)
com.android.internal.os.ZygoteInit.main (ZygoteInit.java:632)
```

### CODE REVIEW

File: src/tsconfig.json [hunk 1]
```diff
@@ -17,22 +17,21 @@
         "noImplicitAny": false,
         "noImplicitReturns": true,
         "noImplicitUseStrict": false,
-        "noFallthroughCasesInSwitch": true
+        "noFallthroughCasesInSwitch": true,
+        "skipLibCheck": true
```
Comment (VladimirAmiorkov) [inline src/tsconfig.json:21]: Is this needed and why, skipping lib check may lead to issues going forward

File: src/tsconfig.json [hunk 2]
```diff
@@ -17,22 +17,21 @@
...
         "./nativescript-fresco-common.ts",
         "./nativescript-fresco.android.ts",
         "./nativescript-fresco.ios.ts",
         "./angular/nativescript-fresco-directives.ts",
         "./angular/nativescript-fresco.module.ts",
-        "./angular/index.ts",
-        "references.d.ts"
-    ],
-    "exclude": [
-        "node_modules"
```
Comment (VladimirAmiorkov) [inline src/tsconfig.json:35]: Why did you remove this exclude? Running tsc on the node_modules should not be necessary. Multiple plugins including the [plugin seed](https://github.com/NativeScript/nativescript-plugin-seed/blob/master/src/tsconfig.json) add the node_modules to the excludes of the tsconfig.

**Review by VladimirAmiorkov — CHANGES_REQUESTED** [review]
I have concerns regarding the skipLibCheck and the removed node_modules in the tsconfing excludes

File: src/tsconfig.json (hunk 1 above)
Comment (DickSmith) [reply src/tsconfig.json:21]: @VladimirAmiorkov 
libCheck often fails for silly reasons such as if multiple libraries have a type called `Log`.
It also adds unnecessary compilation time and sometimes fails when using newer version of TS in a way that has no impact on actual usage.
All it really does is check that other libs are using other lib's types correctly/currently, and really doesn't have any usefulness when consuming them.

File: src/tsconfig.json (hunk 2 above)
Comment (DickSmith) [reply src/tsconfig.json:35]: @VladimirAmiorkov 
When using `include` it isn't necessary to exclude `node_modules`, since it isn't in the `include`, so therefore won't be included. The only time you'd want to use include and exclude was if your include had something like `libs/**` and then your exclude had `libs/xyz` because maybe you have lots of stuff in `libs` that you don't want to list out in the `include` but you want `libs/xyz` to be excluded for some reason.

I know `plugin seed` and many others use `exclude`, but usually that's just because they have no `include`, so the default behaviour is to include everything within the directory so you just filter out `node_modules`, but if you're already using `include` then excluding `node_modules` is redundant and can be misleading.

For ease of use I think `plugin seed` and many others just use `exclude` because then you don't need to think about what files need to be compiled, but if you have `include` you already have, and `include` is much more powerful and scoped and less likely to result in strange behaviour in the long run.

*tl;dr*: because `include` is already being used and `node_modules` isn't in there, it is already being excluded from the `tsc`.

**Comment by VladimirAmiorkov** [comment]
Hi @DickSmith ,

Thank you for your PR. As I was in the middle of migrating (#16) the entire plugin to use the latest version of the native Android library I also internally applied your PR changes as it would have made this PR outdated and require it to be rebased. 

I will be closing [this](https://github.com/NativeScript/nativescript-fresco/pull/77) PR you can follow this one to be notified when it is released with your changes.

File: src/tsconfig.json (hunk 2 above)
Comment (VladimirAmiorkov) [reply src/tsconfig.json:35]: Yeah, I removed the includes in the new PR that will bring the latest native lib to the plugin for this reason. It looks like when this plugin was developer back in the day the decision to use include was taken but I don't see ant reason to use it rather than use the exclude option.
